ddtrace/tracer.py

```diff
@@ -694,6 +694,9 @@ def _start_span(
             span._local_root = span
             if config.report_hostname:
                 span.set_tag_str(HOSTNAME_KEY, hostname.get_hostname())
+
+            if config.env:
```

Comment:
Nice catch 😄. We also set the env tag on line 743. Can we set this tag once? (maybe on line 662)

Reply:
I tried that, but doing so breaks a ton of tracer tests. I decided that this duplicating approach was the best balance between safety and fixing this bug.

Reply:
Your previous [change](https://github.com/DataDog/dd-trace-py/pull/5339/commits/bddad22dd8824ca858869466e24f6a05d13da251) caused 3 test cases to fail, just on multiple versions being tested, so that's why it looks like a ton of tests failed. I think it's worth at least investigating why that change causes it to fail before we can accept the duplicate code 🤔